### PR TITLE
Implements vanilla StatsD backend

### PR DESCRIPTION
**Pull request by jruere**
Implemented as discussed in #3.

### CODE REVIEW

File: markus/backends/statsd.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      Defaults to ``None``.
+
+    * statsd_maxudpsize: the maximum data to send per packet
+
+      Defaults to ``512``.
+
+
+    .. seealso::
+
+       http://docs.datadoghq.com/guides/metrics/
```
Comment (willkg) [inline markus/backends/statsd.py]: This seealso should probably be a different url so people who follow it don't get confused by Datadog statsd enhancements.

Maybe this url?: http://statsd.readthedocs.io/en/v3.2.1/configure.html

File: markus/backends/statsd.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    def gauge(self, stat, value, tags=None):
+        """Set a gauge"""
+        self.client.gauge(stat=stat, value=value)
+
+    def timing(self, stat, value, tags=None):
+        """Measure a timing for statistical distribution"""
+        self.client.timing(stat=stat, delta=value)
+
+    def histogram(self, stat, value, tags=None):
+        """Measure a value for statistical distribution"""
```
Comment (willkg) [inline markus/backends/statsd.py]: This should mention that it's turning around and using the timing, too. Kind of like how I did this:

https://github.com/willkg/markus/blob/fe2d5a365beb2a4dfa650c1790332c03ee14c376/markus/backends/cloudwatch.py#L61

File: setup.py [hunk 3]
```diff
@@ -54,7 +54,8 @@ def get_file(fn):
         'six',
     ],
     extra_requires={
-        'datadog': ['datadog']
+        'datadog': ['datadog'],
+        'statsd': ['statd'],
```
Comment (willkg) [inline setup.py]: I think this should be;

```
'statsd': ['statsd'],
```

File: tests/test_statsd.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+def test_histogram(mockstatsd):
+    ddm = statsd.StatsdMetrics({})
+
+    ddm.histogram('foo', value=4321, tags=['key1:val'])
+
+    assert (
+        ddm.client.calls ==
+        [('timing', (), {'stat': 'foo', 'delta': 4321})]
+    )
```
Comment (willkg) [inline tests/test_statsd.py]: These tests are great!

Having said that, if you have better ideas for how to test backends in Markus, I'm all ears.

**Comment by willkg** [comment]
I had a couple of minor issues, but otherwise this is great! If you could fix those, I'd love to merge this.

Thank you!

**Comment by willkg** [comment]
@jruere How's this coming along? Is there anything I can do to help you land it?

**Comment by jruere** [comment]
Sorry, I'm quite busy ATM and I could not use the project internally in the end.

I'm not sure when I'll resume this work.

**Comment by willkg** [comment]
That's totally fine! I can take this and finish it up. Thank you for working on it!

**Comment by willkg** [comment]
I took what you did and then finished it up. That's in pull request #22.

Thank you for working on this!

**Pull request closed without merging**